### PR TITLE
Test

### PR DESCRIPTION
**Pull request by SherryLiu3670**
# fig, axs = plt.subplots(2, 2, figsize=(15, 5))
    # axs[0][0].plot(statistics["train_loss"])
    # axs[0][0].set_title("Train loss")
    # axs[1][0].plot(statistics["train_accuracy"])
    # axs[1][0].set_title("Train accuracy")
    # axs[0][1].plot(statistics["validation_loss"])
    # axs[0][1].set_title("Validation loss")
    # axs[1][1].plot(statistics["validation_accuracy"])
    # axs[1][1].set_title("Validation accuracy")
    # fig.savefig("../../../reports/figures/training_statistics.png")
cancel saving this picture locally, the path is a relative path that will cause test fail.

### CODE REVIEW

File: src/food_classification/data.py [hunk 1]
```diff
@@ -18,7 +17,7 @@ def __init__(self, **preprocessed_dict) -> None:
         output_folder = preprocessed_dict["output_folder"]
         train_images_file = preprocessed_dict["train_images_file"]
         train_target_file = preprocessed_dict["train_target_file"]
-        project_root = get_original_cwd()
+        project_root = os.getcwd()
```
Comment (umairmukati) [inline src/food_classification/data.py:20]: Does this return the same path as get_original_cwd()?

File: src/food_classification/data.py [hunk 2]
```diff
@@ -48,14 +47,14 @@ class MNISTTestDataset(Dataset):
     def __init__(self, **preprocessed_dict) -> None:
 
         output_folder = preprocessed_dict["output_folder"]
-        test_images_file = preprocessed_dict["test_images_file"]
-        test_target_file = preprocessed_dict["test_target_file"]
-        project_root = get_original_cwd()
+        test_images_file = preprocessed_dict["train_images_file"]
```
Comment (umairmukati) [inline src/food_classification/data.py:50]: Is this a mistake?

File: src/food_classification/data.py [hunk 3]
```diff
@@ -48,14 +47,14 @@ class MNISTTestDataset(Dataset):
     def __init__(self, **preprocessed_dict) -> None:
 
         output_folder = preprocessed_dict["output_folder"]
-        test_images_file = preprocessed_dict["test_images_file"]
-        test_target_file = preprocessed_dict["test_target_file"]
-        project_root = get_original_cwd()
+        test_images_file = preprocessed_dict["train_images_file"]
+        test_target_file = preprocessed_dict["train_target_file"]
```
Comment (umairmukati) [inline src/food_classification/data.py:51]: Same here. Mistake?

File: src/food_classification/train.py [hunk 4]
```diff
@@ -2,10 +2,8 @@
 import matplotlib.pyplot as plt
 import torch
 import typer
-import data
-# from . import data
-# from . import model as models
-
+import food_classification.data as data
```
Comment (umairmukati) [inline src/food_classification/train.py:5]: Why is it required?

File: tests/test_train.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    model = torch.nn.Module()
+    cfg.experiment.device = "cuda" if torch.cuda.is_available() else "cpu"
+    try:
+        model = hydra.utils.instantiate(cfg.model).to(cfg.experiment.device)
+        assert model is not None, "Model should be initialized successfully."
+    except Exception as e:
+        pytest.fail(f"Model initialization failed: {e}")
+
+
+def test_data_loading(cfg):
```
Comment (umairmukati) [inline tests/test_train.py:40]: Shouldnt this be in the test_data.py file?

File: src/food_classification/data.py (hunk 1 above)
Comment (SherryLiu3670) [reply src/food_classification/data.py:20]: yes, as we don't use hydra decorator before this function, we don't need to use get_original_cwd(), this function is only used if we use hydra

File: src/food_classification/data.py (hunk 3 above)
Comment (SherryLiu3670) [reply src/food_classification/data.py:51]: Yes, it's the same thing

File: tests/test_train.py (hunk 5 above)
Comment (SherryLiu3670) [reply tests/test_train.py:40]: Yes, but test data is not my part. I wrote this in case the dataloader fails leading to the failure of training. (cause I don't have test_data.py on my side) We can just delete this.
